### PR TITLE
PLT-3492 Fix connection leaks in push notifications, diagnostics and oauth

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
Fix the last file descriptor leaks, these ones caused by HTTP requests made in push notifications, diagnostics and OAuth.

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-3492

### CODE REVIEW

**Comment by enahum** [comment]
+1

File: api/oauth.go [hunk 1]
```diff
@@ -195,6 +196,11 @@ func completeOAuth(c *Context, w http.ResponseWriter, r *http.Request) {
 		c.Err = err
 		return
 	} else {
+		defer func() {
+			ioutil.ReadAll(body)
```
Comment (crspeller) [inline api/oauth.go]: Why is it necessary to ReadAll before closing? Shouldn't Close be enough?

Comment (jwilander) [inline api/oauth.go]: Unless the body is read completely first the connection resource won't release right away. It eventually does but this makes sure it's released immediately

**Comment by coreyhulen** [comment]
+1

**Pull request merged**